pkg/controller/acmeorders/sync_test.go

```diff
@@ -90,7 +90,12 @@ func TestSync(t *testing.T) {
 		}),
 	)
 
-	testOrderIP := gen.Order("testorder", gen.SetOrderIssuer(cmmeta.ObjectReference{Name: testIssuerHTTP01.Name}), gen.SetOrderIPAddresses("10.0.0.1"))
+	testOrderIP := gen.Order("testorder",
+		gen.SetOrderCommonName("10.0.0.2"),
```

Comment:
Maybe also assert that the IP from CN is added to the IPs? I am not too familiar with the test structure here, but it seems like it should be possible around here:

https://github.com/cert-manager/cert-manager/blob/833e97bdf2285403a2a876d6c4d0a9c6b9a4261c/pkg/controller/acmeorders/sync_test.go#L411-L413